### DISCUSSION TITLE
use `{thing}` for string interpolation

### DISCUSSION THREAD

**Discussion by Hunam6** [vlang/v#15063] [Ideas]
As discussed [here](https://canary.discord.com/channels/592103645835821068/700746775962714232/873194542856290315) & [here](https://github.com/vlang/v/issues/14804#issuecomment-1161932772), we should switch to only one consistent string interpolation system, `{thing}` seems ideal.

**Comment by yuyi98** [comment]
How to handle this situation appropriately?
```v
pub fn str_intp_sq(in_str string) string {
	return 'str_intp(2, _MOV((StrIntpData[]){{_SLIT("\'"), $si_s_code, {.d_s = $in_str}},{_SLIT("\'"), 0, {.d_c = 0 }}}))'
}
```

**Comment by Hunam6** [comment]
That's a tricky one for sure..

I would say either escape all `{` & `}`.

```v
pub fn str_intp_sq(in_str string) string {
	return 'str_intp(2, _MOV((StrIntpData[])\{\{_SLIT("\'"), {si_s_code}, \{.d_s = {in_str}\}\},\{_SLIT("\'"), 0, \{.d_c = 0 \}\}\}))'
}
```

either use string concatenation combined with raw strings

```v
pub fn str_intp_sq(in_str string) string {
	return 'str_intp(2, _MOV((StrIntpData[]){{_SLIT("\'"), ' + si_s_code + ', {.d_s = ' + in_str + r'}},{_SLIT("\'"), 0, {.d_c = 0 }}}))'
}
```
